### DISCUSSION TITLE
Automatically Remove Underline for Frequently Used Words

### DISCUSSION THREAD

**Discussion by fanurs** [streetsidesoftware/vscode-spell-checker#3517] [New Feature Requests]
I have tried to check whether this feature is supported, but I could not find it. Apologies if it is already available.

I would like to request an opt-in feature where a word that is used frequently (e.g., 5 times or more) within a document or workspace would automatically have all its underlines removed. The rationale is that if a user consistently uses a word multiple times, it indicates an intentional usage, and the underline can be safely dropped.

**Details:**
- **Usage Threshold:** Allow users to set a threshold (e.g., 5 occurrences) after which the underline for a specific word is removed within the document or workspace.
- **Opt-in Feature:** This feature should be disabled by default. Users can enable it and set their preferred threshold if desired.

Thank you for considering this feature request!

**Comment by Jason3S** [comment]
@fanurs,

Thank you for the suggestion.

What problem are you trying to solve? I'm asking because I'm trying to figure out why adding the word to your dictionary doesn't work.

**Comment by fanurs** [comment]
Adding a word to the dictionary works, but it can be a little tedious. The feature I'm suggesting is more dynamic - automatically removing the underline after a word is used frequently within a specific document (opt-in feature, of course). This would be especially helpful when working with existing codebases or pasted code, where repeatedly underlined terms clearly indicate intended usage.

If needed, I have experience developing VS Code extensions in TypeScript with a small userbase (10k+). If someone could point me to the relevant files, I’d be happy to contribute and help test this feature.

Thanks again for considering this.
